src/core_plugins/interpreter/public/functions/kibana_context.js

```diff
@@ -0,0 +1,89 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import chrome from 'ui/chrome';
+import { i18n } from '@kbn/i18n';
+
+export default () => ({
+  name: 'kibana_context',
+  type: 'kibana_context',
+  context: {
+    types: [
+      'kibana_context',
+      'null',
+    ],
+  },
+  help: i18n.translate('common.core_plugins.interpreter.public.functions.kibana_context.help', {
+    defaultMessage: 'Updates kibana global context'
+  }),
+  args: {
+    q: {
+      types: ['string', 'null'],
+      aliases: ['query', '_'],
+      default: null,
+    },
+    filters: {
+      types: ['string', 'null'],
+      default: '"[]"',
+    },
+    timeRange: {
+      types: ['string', 'null'],
+      default: null,
+    },
+    savedSearchId: {
+      types: ['string', 'null'],
+      default: null,
+    }
+  },
+  fn(context, args) {
+    return chrome.dangerouslyGetActiveInjector().then(async $injector => {
```

Comment:
For better readability I would suggest, that we're also using `await` here (and mark `fn` as `async`) as in the rest of the function, i.e.

```js
const $injector = await chrome.dangerouslyGetActiveInjector();
```

I would suggest the same for all other functions.